jenkins-job-generator/definitions/build-jobs.yml

```diff
@@ -531,13 +531,35 @@
                         build job: 'generate-cf-java-buildpack-data', propagate: false, wait: false
                     }}
                 }}
+                stage('Build Debian Linux Packages') {{
+                    when {{
+                        expression {{ env.VERIFICATION_RESULT != "2" && params.PUBLISH == true && params.RELEASE == true }}
+                    }}
+                    stage("Checkout Infrastructure Repository") {
+                        steps {
+                            checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'SapMachine-Infrastructure']], userRemoteConfigs: [[credentialsId: 'SapMachine-github', url: 'https://github.com/SAP/SapMachine-infrastructure.git']]]
+                        }
+                    }
+                    stage('Build') {
+                        steps {
+                            sh 'rm -f *.deb'
+                            sh "python3 SapMachine-Infrastructure/lib/make_deb.py --tag=${params.GIT_TAG_NAME} --templates-directory=SapMachine-Infrastructure/debian-templates"
+                        }
+                        post {
+                            success {
+                                archiveArtifacts allowEmptyArchive: true, artifacts: "*.deb"
+                            }
+                        }
+                    }
+                }}
                 stage('Publish Debian Linux Packages') {{
                     when {{
-                        expression {{ env.VERIFICATION_RESULT != "2" && params.PUBLISH == true && params.RELEASE == true && JOB_NAME ==~ /((\S*)(release-linux_x86_64)(\S*))/ }}
+                        expression {{ env.VERIFICATION_RESULT != "2" && params.PUBLISH == true && params.RELEASE == true }}
```

Comment:
We still need the check for linux_x86_64 job at the moment, since it's the only supported platform for debian packages.